feat(tutorials): add first version of tutorial about ontology (#272)
okp4/docs: docs/tutorials/ontology-1.md

---
sidebar_position: 6
---

# Leverage the ontology

![Tutorials overview](/img/content/tutorials/ontology-1.webp)

> This tutorial will unravel the concept of ontology and figure out its crucial role in the OKP4 protocol.

Reviewer: ```suggestion
This tutorial unveils the concept of ontology and figure out its crucial role in the OKP4 protocol.
```

> An ontology describes knowledge. We can represent it as a graph database that fixes the meaning of concepts and the relationships that unite them.

Reviewer: ```suggestion
An ontology describes knowledge. We can represent it as a graph that fixes the meaning of concepts and the relationships that unite them.
```

### The OKP4 ontology

The [OKP4 ontology](https://ontology.okp4.space) describes and defines the different forms of vocabulary used in the OKP4 protocol in a standard and well-designed format. You notably should follow the OKP4 ontology to reference resources (services, datasets, …) you want to share.  

:::info

Check out [the OKP4 ontology GitHub repository](https://github.com/okp4/ontology) to get all [the different OKP4 specifications](https://github.com/okp4/ontology/tree/main/src) and [some examples](https://github.com/okp4/ontology/tree/main/example).

You can read [this article about the OKP4 ontology](https://blog.okp4.network/okp4-knowledge-sharing-through-ontology-driven-dataverse-a-new-approach-to-knowledge-description-f9b81d4484c8) from the blog to dig into the details.

:::

## **Semantic data expression with RDF triples**

### Creating meaningful relationships

In the realm of the semantic web, RDF (Resource Description Framework) is a common standard used to express these relationships.

RDF encodes information in triple structures. The basic structure of an RDF triple is subject-predicate-object, much like a simple sentence in the English language.

> 1. **Subject**: The subject is the entity or resource the statement is about. It's typically a URI (Uniform Resource Identifier) which uniquely identifies a resource.

Reviewer: ```suggestion
1. **Subject**: The subject is the entity or resource the statement is about. It's typically a URI ([Uniform Resource Identifier](https://en.wikipedia.org/wiki/Uniform_Resource_Identifier)) which uniquely identifies a resource.
```

2. **Predicate**: The predicate (also called a property) is a specific aspect, characteristic, attribute, or relation that describes the subject. It's also typically a URI.

> 3. **Object**: The object is the value of the attribute defined by the predicate for the subject. It can be a URI or a literal (like a string or a number).

Reviewer: ```suggestion
3. **Object**: The object is the value of the attribute defined by the predicate for the subject. It can be a URI or a literal (such as a string or a number) and may also include additional information such as a language tag or a datatype.
```

> In RDF, **prefixes** are used as a shorthand notation for long URIs to make the data more readable and less verbose. They're similar to namespaces in programming languages. For instance, instead of writing `http://www.w3.org/2001/XMLSchema#integer`, you could declare a prefix `xsd` to represent the `http://www.w3.org/2001/XMLSchema` URI and then use `xsd:integer`.

Reviewer: ```suggestion
In RDF, **prefixes** are used as a shorthand notation for long URIs to make the data more readable and less verbose. They're similar to namespaces in programming languages. For instance, instead of writing `http://www.w3.org/2001/XMLSchema#integer`, you could declare a prefix `xsd` to represent the `http://www.w3.org/2001/XMLSchema#` URI and then use `xsd:integer`.
```

| jq '.data.results.bindings'

: '
[
  {
    "serviceId": {
      "type": "uri",
      "value": {
        "full": "https://ontology.okp4.space/dataverse/service/03230ce7-d8cb-410e-919e-19c480c1dd75"
      }
    }
  }
]
'
```

## Recap’

- We need to store data within the OKP4 blockchain to reference resources and their relationships with sharing rules, and service executions, among other pieces of information. The structuration is defined with an ontology.
- The `cognitarium` smart contract enables the storage and query of semantic data, expressed with RDF triples (subject-predicate-object). It's a relevant solution for the needed ontology expressiveness.
- Various formats exist for RDF triples; Turtle, RDF/XML, NTriples/NQuads are those supported.

As you embark on your journey with OKP4, you're crafting the future of semantic web-enabled decentralized systems. And that is a frontier worth exploring!

> The marriage between the blockchain with smart contract technology and ontology isn't just a fancy trend; it's a transformative union that paves the way for novel applications, greater efficiency, and unprecedented interoperability.

Reviewer 2: Mind blowing 😍

Author: :D
